File: Cores/DotNet/Random/Distributions/DiscreteUniform.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace OSOL.Extremum.Cores.DotNet.Random.Distributions
{
    public interface IDiscreteUniform
    {
        int GetDiscreteUniform(int min, int max);   
    }

    public static class DiscreteUniformFunctions
    {
        public static Dictionary<string, int> GetDiscreteUniformVector(this IDiscreteUniform GoRN, Dictionary<string, Tuple<int, int>> area) =>
            area.ToDictionary(kvp => kvp.Key,
                kvp => GoRN.GetDiscreteUniform(kvp.Value.Item1, kvp.Value.Item2));
    }

}

File: Cores/JVM/src/main/scala/OSOL/Extremum/Cores/JVM/Random/Distributions/DiscreteUniform.scala
package OSOL.Extremum.Cores.JVM.Random.Distributions

trait DiscreteUniform {

  def getDiscreteUniform(min: java.lang.Integer, max: java.lang.Integer): java.lang.Integer

  final def getDiscreteUniform(area: Map[String, (java.lang.Integer, java.lang.Integer)]): Map[String, java.lang.Integer] =
    area.map { case (key, (min, max)) => (key, getDiscreteUniform(min, max)) }

  def getDiscreteUniformScala(min: Int, max: Int): Int =
    getDiscreteUniform(min, max).intValue()

  final def getDiscreteUniformScala(area: Map[String, (Int, Int)]): Map[String, Int] =
    area.map { case (key, (min, max)) => (key, getDiscreteUniformScala(min, max)) }

}